<a href="https://colab.research.google.com/github/VeroScoCian/Proyecto-final/blob/main/ProyectoFinal.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Extraccion, Analisis y Gestion de datos para InnovateTech


##Instalo librerias

In [ ]:
!pip install pandas==2.2.2



     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 13.0/13.0 MB 37.2 MB/s eta 0:00:00
  Attempting uninstall: pandas
    Found existing installation: pandas 2.0.3
    Uninstalling pandas-2.0.3:
      Successfully uninstalled pandas-2.0.3
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
cudf-cu12 24.4.1 requires pandas<2.2.2dev0,>=2.0, but you have pandas 2.2.2 which is incompatible.
google-colab 1.0.0 requires pandas==2.0.3, but you have pandas 2.2.2 which is incompatible.


In [ ]:
!pip install numpy

In [ ]:
!pip install openpyxl

In [ ]:
!pip install pyarrow

In [ ]:
!pip install fastparquet

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.7/1.7 MB 7.2 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.0/2.0 MB 16.6 MB/s eta 0:00:00


In [ ]:
!pip install lxml

In [ ]:
import pandas as pd
import numpy as np
import json
import os

## Carga de archivos CSV en DataFrame de Pandas

Cargamos archivos originales para luego analizarlos.

##Customers

In [ ]:
#Importo Dataset como DataFrame, muestro 5 filas para verificar
df_customers = pd.read_csv("https://raw.githubusercontent.com/VeroScoCian/Proyecto-final/main/ecommerce_customers_dataset.csv")
df_customers.head()

customer_id                customer_unique_id  \
0  06b8999e2fba1a1fbc88172c00ba8bc7  861eff4711a542e4b93843c6dd7febb0   
1  18955e83d337fd6b2def6b18a428ac77  290c77bc529b7ac935b93aa66c333dc3   
2  4e7b3e00288586ebd08712fdd0374a03  060e732b5b29e8181a18229c7b0b2b5e   
3  b2b6027bc5c5109e529d4dc6358b12c3  259dac757896d24d7702b9acbbff3f3c   
4  4f2d8ab171c80ec8364f7c12e35b23ad  345ecd01c38d18a9036ed96c73b8d066   

   customer_zip_code_prefix          customer_city customer_state  
0                     14409                 franca             SP  
1                      9790  sao bernardo do campo             SP  
2                      1151              sao paulo             SP  
3                      8775        mogi das cruzes             SP  
4                     13056               campinas             SP

##Order items

In [ ]:
#Importo Dataset como DataFrame, muestro 5 filas para verificar
df_order_items = pd.read_csv("https://raw.githubusercontent.com/VeroScoCian/tpf/main/ecommerce_order_items_dataset.csv")
df_order_items.head()

order_id  order_item_id  \
0  00010242fe8c5a6d1ba2dd792cb16214              1   
1  00018f77f2f0320c557190d7a144bdd3              1   
2  000229ec398224ef6ca0657da4fc703e              1   
3  00024acbcdf0a6daa1e931b038114c75              1   
4  00042b26cf59d7ce69dfabb4e55b4fd9              1   

                         product_id                         seller_id  \
0  4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   
1  e5f2d52b802189ee658865ca93d83a8f  dd7ddc04e1b6c2c614352b383efe2d36   
2  c777355d18b72b67abbeef9df44fd0fd  5b51032eddd242adc84c38acab88f23d   
3  7634da152a4610f1595efa32f14722fc  9d7a1d34a5052409006425275ba1c2b4   
4  ac6c3623068f30de03045865e4e10089  df560393f3a51e74553ab94004ba5c87   

   shipping_limit_date   price  freight_value  
0  2017-09-19 09:45:35   58.90          13.29  
1  2017-05-03 11:05:13  239.90          19.93  
2  2018-01-18 14:48:30  199.00          17.87  
3  2018-08-15 10:10:18   12.99          12.79  
4  2017-02-13 13:57:51  199.90          18.14

##Orders

In [ ]:
#Importo Dataset comoDataFrame, muestro 5 filas para verificar
df_order = pd.read_csv("https://raw.githubusercontent.com/VeroScoCian/Proyecto-final/main/ecommerce_orders_dataset.csv")
df_order.head()

order_id                       customer_id  \
0  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
1  53cdb2fc8bc7dce0b6741e2150273451  b0830fb4747a6c6d20dea0b8c802d7ef   
2  47770eb9100c2d0c44946d9cf07ec65d  41ce2a54c0b03bf3443c3d931a367089   
3  949d5b44dbf5de918fe9c16f97b45f8a  f88197465ea7920adcdbec7375364d82   
4  ad21c59c0840e6cb83a9ceb5573f8159  8ab97904e6daea8866dbdbc4fb7aad2c   

  order_status order_purchase_timestamp    order_approved_at  \
0    delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   
1    delivered      2018-07-24 20:41:37  2018-07-26 03:24:27   
2    delivered      2018-08-08 08:38:49  2018-08-08 08:55:23   
3    delivered      2017-11-18 19:28:06  2017-11-18 19:45:59   
4    delivered      2018-02-13 21:18:39  2018-02-13 22:20:29   

  order_delivered_carrier_date order_delivered_customer_date  \
0          2017-10-04 19:55:00           2017-10-10 21:25:13   
1          2018-07-26 14:31:00           2018-08-07 15:27:45   
2          2018-08-08 13:50:00           2018-08-17 18:06:29   
3          2017-11-22 13:39:59           2017-12-02 00:28:42   
4          2018-02-14 19:46:34           2018-02-16 18:17:02   

  order_estimated_delivery_date  
0           2017-10-18 00:00:00  
1           2018-08-13 00:00:00  
2           2018-09-04 00:00:00  
3           2017-12-15 00:00:00  
4           2018-02-26 00:00:00

##Payments

In [ ]:
#Importo Dataset comoDataFrame, muestro 5 filas para verificar
df_payments= pd.read_csv("https://raw.githubusercontent.com/VeroScoCian/Proyecto-final/main/ecommerce_order_payments_dataset.csv")
df_payments.head()

order_id  payment_sequential payment_type  \
0  b81ef226f3fe1789b1e8b2acac839d17                   1  credit_card   
1  a9810da82917af2d9aefd1278f1dcfa0                   1  credit_card   
2  25e8ea4e93396b6fa0d3dd708e76c1bd                   1  credit_card   
3  ba78997921bbcdc1373bb41e913ab953                   1  credit_card   
4  42fdf880ba16b47b59251dd489d4441a                   1  credit_card   

   payment_installments  payment_value  
0                     8          99.33  
1                     1          24.39  
2                     1          65.71  
3                     8         107.78  
4                     2         128.45

##Products

In [ ]:
#Importo Dataset comoDataFrame, muestro 5 filas para verificar
df_products = pd.read_csv("https://raw.githubusercontent.com/VeroScoCian/Proyecto-final/main/ecommerce_products_dataset.csv")
df_products.head()

product_id  product_category_name  \
0  1e9e8ef04dbcff4541ed26657ea517e5             perfumaria   
1  3aa071139cb16b67ca9e5dea641aaa2f                  artes   
2  96bd76ec8810374ed1b65e291975717f          esporte_lazer   
3  cef67bcfe19066a932b7673e239eb23d                  bebes   
4  9dc1a7de274444849c219cff195d0b71  utilidades_domesticas   

   product_name_lenght  product_description_lenght  product_photos_qty  \
0                 40.0                       287.0                 1.0   
1                 44.0                       276.0                 1.0   
2                 46.0                       250.0                 1.0   
3                 27.0                       261.0                 1.0   
4                 37.0                       402.0                 4.0   

   product_weight_g  product_length_cm  product_height_cm  product_width_cm  
0             225.0               16.0               10.0              14.0  
1            1000.0               30.0               18.0              20.0  
2             154.0               18.0                9.0              15.0  
3             371.0               26.0                4.0              26.0  
4             625.0               20.0               17.0              13.0

#Datos DataFrame

Analizaremos composicion, tamaño, tipo de datos y cantidad de nulos de cada DataFrame.

### Funcion datos *DataFrame*

In [ ]:
#Funcion para cantidad y tipos de datos. Luego la utilizaremos para cada DataFrame
def datosDataFrame(df):
  print("Tamaño de DataFrame:")
  print(df.shape) #Pedimos que imprima tamaño
  print("---------")
  print("Analisis de tipo de Datos:")
  print(df.info()) #Pedimos que imprima el tipo de datos
  print("---------")
  print("Recuento de nulos:")
  print(df.isnull().sum()) #Analizamos nulos
  print("---------")
  print("Recuento de duplicados:")
  print(df.duplicated().sum()) #Analizamos duplicados


###Customers

In [ ]:
#Aplicamos la funcion
datosDataFrame(df_customers)


Tamaño de DataFrame:
(99441, 5)
---------
Analisis de tipo de Datos:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 5 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   customer_id               99441 non-null  object
 1   customer_unique_id        99441 non-null  object
 2   customer_zip_code_prefix  99441 non-null  int64 
 3   customer_city             99441 non-null  object
 4   customer_state            99441 non-null  object
dtypes: int64(1), object(4)
memory usage: 3.8+ MB
None
---------
Recuento de nulos:
customer_id                 0
customer_unique_id          0
customer_zip_code_prefix    0
customer_city               0
customer_state              0
dtype: int64
---------
Recuento de duplicados:
0


###Order Items

In [ ]:
#Aplicamos la funcion
datosDataFrame(df_order_items)

Tamaño de DataFrame:
(112650, 7)
---------
Analisis de tipo de Datos:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 112650 entries, 0 to 112649
Data columns (total 7 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   order_id             112650 non-null  object 
 1   order_item_id        112650 non-null  int64  
 2   product_id           112650 non-null  object 
 3   seller_id            112650 non-null  object 
 4   shipping_limit_date  112650 non-null  object 
 5   price                112650 non-null  float64
 6   freight_value        112650 non-null  float64
dtypes: float64(2), int64(1), object(4)
memory usage: 6.0+ MB
None
---------
Recuento de nulos:
order_id               0
order_item_id          0
product_id             0
seller_id              0
shipping_limit_date    0
price                  0
freight_value          0
dtype: int64
---------
Recuento de duplicados:
0


###Orders

In [ ]:
#Aplicamos la funcion
datosDataFrame(df_order)

Tamaño de DataFrame:
(99441, 8)
---------
Analisis de tipo de Datos:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 8 columns):
 #   Column                         Non-Null Count  Dtype 
---  ------                         --------------  ----- 
 0   order_id                       99441 non-null  object
 1   customer_id                    99441 non-null  object
 2   order_status                   99441 non-null  object
 3   order_purchase_timestamp       99441 non-null  object
 4   order_approved_at              99281 non-null  object
 5   order_delivered_carrier_date   97658 non-null  object
 6   order_delivered_customer_date  96476 non-null  object
 7   order_estimated_delivery_date  99441 non-null  object
dtypes: object(8)
memory usage: 6.1+ MB
None
---------
Recuento de nulos:
order_id                            0
customer_id                         0
order_status                        0
order_purchase_timestamp            0
order_a

###Payments

In [ ]:
#Aplicamos la funcion
datosDataFrame(df_payments)

Tamaño de DataFrame:
(103886, 5)
---------
Analisis de tipo de Datos:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 103886 entries, 0 to 103885
Data columns (total 5 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   order_id              103886 non-null  object 
 1   payment_sequential    103886 non-null  int64  
 2   payment_type          103886 non-null  object 
 3   payment_installments  103886 non-null  int64  
 4   payment_value         103886 non-null  float64
dtypes: float64(1), int64(2), object(2)
memory usage: 4.0+ MB
None
---------
Recuento de nulos:
order_id                0
payment_sequential      0
payment_type            0
payment_installments    0
payment_value           0
dtype: int64
---------
Recuento de duplicados:
0


###Products

In [ ]:
#Aplicamos la funcion
datosDataFrame(df_products)

Tamaño de DataFrame:
(32951, 9)
---------
Analisis de tipo de Datos:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32951 entries, 0 to 32950
Data columns (total 9 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   product_id                  32951 non-null  object 
 1   product_category_name       32341 non-null  object 
 2   product_name_lenght         32341 non-null  float64
 3   product_description_lenght  32341 non-null  float64
 4   product_photos_qty          32341 non-null  float64
 5   product_weight_g            32949 non-null  float64
 6   product_length_cm           32949 non-null  float64
 7   product_height_cm           32949 non-null  float64
 8   product_width_cm            32949 non-null  float64
dtypes: float64(7), object(2)
memory usage: 2.3+ MB
None
---------
Recuento de nulos:
product_id                      0
product_category_name         610
product_name_lenght           610
product_

# Transformacion DataFrame
En esta seccion del codigo nos encargamos de realizar funciones correspondientes para evitar posibles problemas a futuro.
Estas funciones nos ayudaran a tener un cnjunto de datos homogeneos, como asi tambien, la eliminacion de ser necesario de filas con datos nulos.

###Funcion transformarFechas

In [ ]:
#Funcion para transformar a datetime
def transformarFechas(df, columnas_fechas):
  df[columnas_fechas] = df[columnas_fechas].apply(pd.to_datetime) #Transformamos la columna_fechas a datetime

###Funcion TransformarString

In [ ]:
#Funcion para transformar a string
def transformarString(df, columnas_string): #Los parametros son el dataframe y la lista con las columnas a transformar
    for columna in columnas_string: #Transforma las columnas en columnas_string a string
        df[columna] = df[columna].astype("string")
    return df

###Funcion TransformarMayusculas

In [ ]:
#Definimos nuestra función convertirMayusculas
def transformarMayusculas(df, columna): #Los parametros son el dataframe y la lista con las columna a transformar
    df[columna] = df[columna].str.upper() # Convertir todos los valores de la columna a mayusculas
    return df


###Funcion Transformar Minusculas

In [ ]:
#Funcion para convertir a Minusculas
def transformarMinusculas(df, columna):
  df[columna] = df[columna].str.lower() #convertimos los valores de la columna a minuscula
  return df

###Customers

In [ ]:
# Lista de columnas a transfomar en string
columnas_string_customers = ["customer_id","customer_unique_id","customer_zip_code_prefix", "customer_city","customer_state"]

#Aplicamos nuestra funcion conversora de string
df_customers = transformarString(df_customers, columnas_string_customers)


In [ ]:
#Revisamos que los tipos de datos sean correctos
df_customers.info()



<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 5 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   customer_id               99441 non-null  string
 1   customer_unique_id        99441 non-null  string
 2   customer_zip_code_prefix  99441 non-null  string
 3   customer_city             99441 non-null  string
 4   customer_state            99441 non-null  string
dtypes: string(5)
memory usage: 3.8 MB


In [ ]:
#Aplicamos la conversion a minusculas
df_customers = transformarMinusculas(df_customers, "customer_city")


In [ ]:
#Aplicamos la conversion a mayusculas
df_customers = transformarMayusculas(df_customers, "customer_state")

###Order Items

In [ ]:
# Lista de columnas que contienen fechas
columnas_fechas_order_items = ['shipping_limit_date']

#Aplicamos nuestra funcion conversora de fechas
df_order_items = transformarFechas(df_order_items, columnas_fechas_order_items)

In [ ]:
# Lista de columnas a transfomar en string
columnas_string_order_items = ["order_id","order_item_id","product_id","seller_id"]

#Aplicamos nuestra funcion conversora de string
df_order_items = transformarString(df_order_items, columnas_string_order_items)

###Orders

In [ ]:
# Porcentaje de nulos en cada columna
null_percentages_orders = (df_order.isnull().sum() / len(df_order)) * 100

#Veamos cual es el % de nulos
print(null_percentages_orders)


order_id                         0.000000
customer_id                      0.000000
order_status                     0.000000
order_purchase_timestamp         0.000000
order_approved_at                0.160899
order_delivered_carrier_date     1.793023
order_delivered_customer_date    2.981668
order_estimated_delivery_date    0.000000
dtype: float64


In [ ]:
# Filtramos las columnas con menos del 10% de valores nulos
columns_to_keep_orders = null_percentages_orders[null_percentages_orders <= 10].index

# Eliminamos las filas con nulos solo en las columnas seleccionadas
df_order.dropna(subset=columns_to_keep_orders, inplace=True)

#Chequeo cantidad de null
df_order.isnull().sum()



order_id                         0
customer_id                      0
order_status                     0
order_purchase_timestamp         0
order_approved_at                0
order_delivered_carrier_date     0
order_delivered_customer_date    0
order_estimated_delivery_date    0
dtype: int64

In [ ]:
# Lista de columnas que contienen fechas
columnas_fechas_orders = ['order_purchase_timestamp', 'order_approved_at',
                          'order_delivered_carrier_date', 'order_delivered_customer_date',
                          'order_estimated_delivery_date']

#Aplicamos nuestra funcion conversora de fechas
df_order = transformarFechas(df_order, columnas_fechas_orders)



In [ ]:
# Lista de columnas a transfomar en string
columnas_string= ["order_id","customer_id","order_status"]

#Aplicamos nuestra funcion conversora de string
df_order = transformarString(df_order, columnas_string)



In [ ]:
#Revisamos que los tipos de datos sean correctos
df_order.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 8 columns):
 #   Column                         Non-Null Count  Dtype 
---  ------                         --------------  ----- 
 0   order_id                       99441 non-null  string
 1   customer_id                    99441 non-null  string
 2   order_status                   99441 non-null  string
 3   order_purchase_timestamp       99441 non-null  object
 4   order_approved_at              99281 non-null  object
 5   order_delivered_carrier_date   97658 non-null  object
 6   order_delivered_customer_date  96476 non-null  object
 7   order_estimated_delivery_date  99441 non-null  object
dtypes: object(5), string(3)
memory usage: 6.1+ MB


In [ ]:
#Aplicamos la conversion a minusculas
df_order = transformarMinusculas(df_order, "order_status")


###Payments

In [ ]:
# Lista de columnas a transfomar en string
columnas_string_payments = ["order_id","payment_type"]

#Aplicamos nuestra funcion conversora de string
df_payments = transformarString(df_payments, columnas_string_payments)


In [ ]:
#Aplicamos la conversion a minusculas
df_payments = transformarMinusculas(df_payments, "payment_type")


###Products

In [ ]:
# Porcentaje de nulos en cada columna
null_percentages_products = (df_products.isnull().sum() / len(df_products)) * 100

print(null_percentages_products)

product_id                    0.000000
product_category_name         1.851234
product_name_lenght           1.851234
product_description_lenght    1.851234
product_photos_qty            1.851234
product_weight_g              0.006070
product_length_cm             0.006070
product_height_cm             0.006070
product_width_cm              0.006070
dtype: float64


In [ ]:
# Filtramos las columnas con menos del 10% de valores nulos
columns_to_keep_products = null_percentages_products[null_percentages_products <= 10].index

# Eliminamos las filas con nulos solo en las columnas seleccionadas
df_products.dropna(subset=columns_to_keep_products, inplace=True)

#Chequeo cantidad de null
df_products.isnull().sum()

product_id                    0
product_category_name         0
product_name_lenght           0
product_description_lenght    0
product_photos_qty            0
product_weight_g              0
product_length_cm             0
product_height_cm             0
product_width_cm              0
dtype: int64

In [ ]:
# Lista de columnas a transfomar en string
columnas_string_products = ["product_id","product_category_name"]

#Aplicamos nuestra funcion conversora de string
df_products = transformarString(df_products, columnas_string_products)


In [ ]:
#Revisamos que los tipos de datos sean correctos
df_products.info()

<class 'pandas.core.frame.DataFrame'>
Index: 32340 entries, 0 to 32950
Data columns (total 9 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   product_id                  32340 non-null  string 
 1   product_category_name       32340 non-null  string 
 2   product_name_lenght         32340 non-null  float64
 3   product_description_lenght  32340 non-null  float64
 4   product_photos_qty          32340 non-null  float64
 5   product_weight_g            32340 non-null  float64
 6   product_length_cm           32340 non-null  float64
 7   product_height_cm           32340 non-null  float64
 8   product_width_cm            32340 non-null  float64
dtypes: float64(7), string(2)
memory usage: 2.5 MB


In [ ]:

#Aplicamos la conversion a minusculas
df_products = transformarMinusculas(df_products, "product_category_name")

#Analisis DataFrame
Una vez realizada la transformacion de datos podemos analizarlos y sacar todos lo necesario para entender mejor a su publico objetivo.

###Funcion para analisis de Dataframes
Realizamos una funcion unica para luego utilizarla con cada nuevo DataFrame

In [ ]:
#Definimos una función para analizar nuestro DataFrame
def analisisDataFrames(df): #Nuestro DataFrame como parametro
  print("----------")
  print("Análisis variables cuantitativas")
  columnas_numericas = df.select_dtypes(include=['int', 'float']).columns #Extraemos las columnas numericas
  if not columnas_numericas.empty: #Aplicamos un if en caso de que no haya columnas numericas
    print(df[columnas_numericas].describe()) #Describe para las columnas cuantitativas
  print("----------")
  print("----------")
  print("Análisis variables cualitativas")
  columnas_string = df.select_dtypes(include=['string']).columns #Extraemos las columnas string
  if not columnas_string.empty: #Aplicamos un if en caso de que no haya columnas del tipo string
    print(df[columnas_string].describe()) #Describe para las columnas cualitativas



#Customers

In [ ]:
#Aplicamos la nueva funcion en cada DataFrame
analisisDataFrames(df_customers)

----------
Análisis variables cuantitativas
----------
----------
Análisis variables cualitativas
                             customer_id                customer_unique_id  \
count                              99441                             99441   
unique                             99441                             96096   
top     06b8999e2fba1a1fbc88172c00ba8bc7  8d50f5eadf50201ccdcedfb9e2ac8455   
freq                                   1                                17   

       customer_zip_code_prefix customer_city customer_state  
count                     99441         99441          99441  
unique                    14994          4119             27  
top                       22790     sao paulo             SP  
freq                        142         15540          41746  


Podemos ver que la cantidad de Clientes unicos en el conjunto de datos es de 96096.

#Order Items

In [ ]:
#Aplicamos la nueva funcion en cada DataFrame
analisisDataFrames(df_order_items)

----------
Análisis variables cuantitativas
               price  freight_value
count  112650.000000  112650.000000
mean      120.653739      19.990320
std       183.633928      15.806405
min         0.850000       0.000000
25%        39.900000      13.080000
50%        74.990000      16.260000
75%       134.900000      21.150000
max      6735.000000     409.680000
----------
----------
Análisis variables cualitativas
                                order_id order_item_id  \
count                             112650        112650   
unique                             98666            21   
top     8272b63d03f5f79c56e9e4120aec44ef             1   
freq                                  21         98666   

                              product_id                         seller_id  
count                             112650                            112650  
unique                             32951                              3095  
top     aca2eb7d00ea1a7b8ebd4e68314663af  6560211a19b479

El promedio de valores de pago por pedidos se encuentra debajo de los 134 Reales. Como podemos ver reflejado en el 75% de las compras.

#Orders


In [ ]:
#Aplicamos la nueva funcion en cada DataFrame
analisisDataFrames(df_order)

----------
Análisis variables cuantitativas
----------
----------
Análisis variables cualitativas
                                order_id                       customer_id  \
count                              99441                             99441   
unique                             99441                             99441   
top     e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
freq                                   1                                 1   

       order_status  
count         99441  
unique            8  
top       delivered  
freq          96478  


El total de pedidos realizados es de 99441 pedidos.

#Payments

In [ ]:
#Aplicamos la nueva funcion en cada DataFrame
analisisDataFrames(df_payments)

----------
Análisis variables cuantitativas
       payment_sequential  payment_installments  payment_value
count       103886.000000         103886.000000  103886.000000
mean             1.092679              2.853349     154.100380
std              0.706584              2.687051     217.494064
min              1.000000              0.000000       0.000000
25%              1.000000              1.000000      56.790000
50%              1.000000              1.000000     100.000000
75%              1.000000              4.000000     171.837500
max             29.000000             24.000000   13664.080000
----------
----------
Análisis variables cualitativas
                                order_id payment_type
count                             103886       103886
unique                             99440            5
top     fa65dad1b0e818e3ccc5cb0e39231352  credit_card
freq                                  29        76795


#Products

In [ ]:
#Aplicamos la nueva funcion en cada DataFrame
analisisDataFrames(df_products)

----------
Análisis variables cuantitativas
       product_name_lenght  product_description_lenght  product_photos_qty  \
count         32340.000000                32340.000000        32340.000000   
mean             48.476592                  771.492393            2.188961   
std              10.245699                  635.124831            1.736787   
min               5.000000                    4.000000            1.000000   
25%              42.000000                  339.000000            1.000000   
50%              51.000000                  595.000000            1.000000   
75%              57.000000                  972.000000            3.000000   
max              76.000000                 3992.000000           20.000000   

       product_weight_g  product_length_cm  product_height_cm  \
count      32340.000000       32340.000000       32340.000000   
mean        2276.956586          30.854545          16.958813   
std         4279.291845          16.955965          13.636

In [ ]:
#calculamos la categoria de productos mas vendida
df_products['product_category_name'].value_counts()

product_category_name
cama_mesa_banho                  3029
esporte_lazer                    2867
moveis_decoracao                 2657
beleza_saude                     2444
utilidades_domesticas            2335
                                 ... 
fashion_roupa_infanto_juvenil       5
casa_conforto_2                     5
pc_gamer                            3
seguros_e_servicos                  2
cds_dvds_musicais                   1
Name: count, Length: 73, dtype: Int64

En este analisis encontramos que la categoria de productos mas vendida es la cama_mesa_banho, con una cantidad de 3029 ventas.

#Nuevos DataFrame
Una vez realizados toda la limpieza, analisis y extraccion, realizaremos 2 nuevos DataFrame que exportaremos a CSV para poder realizar el reporte adecuado.

In [ ]:
# Crear la carpeta 'dataset_generados'
if not os.path.exists('dataset_generados'):
    os.makedirs('dataset_generados')

#Orders - Payments

In [ ]:
#Realizamos un merge
df_merge_orders_payments = pd.merge(df_order, df_payments, how = "inner", on= "order_id")
df_merge_orders_payments.head()

order_id                       customer_id  \
0  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
1  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
2  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
3  53cdb2fc8bc7dce0b6741e2150273451  b0830fb4747a6c6d20dea0b8c802d7ef   
4  47770eb9100c2d0c44946d9cf07ec65d  41ce2a54c0b03bf3443c3d931a367089   

  order_status order_purchase_timestamp    order_approved_at  \
0    delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   
1    delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   
2    delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   
3    delivered      2018-07-24 20:41:37  2018-07-26 03:24:27   
4    delivered      2018-08-08 08:38:49  2018-08-08 08:55:23   

  order_delivered_carrier_date order_delivered_customer_date  \
0          2017-10-04 19:55:00           2017-10-10 21:25:13   
1          2017-10-04 19:55:00           2017-10-10 21:25:13   
2          2017-10-04 19:55:00           2017-10-10 21:25:13   
3          2018-07-26 14:31:00           2018-08-07 15:27:45   
4          2018-08-08 13:50:00           2018-08-17 18:06:29   

  order_estimated_delivery_date  payment_sequential payment_type  \
0           2017-10-18 00:00:00                   1  credit_card   
1           2017-10-18 00:00:00                   3      voucher   
2           2017-10-18 00:00:00                   2      voucher   
3           2018-08-13 00:00:00                   1       boleto   
4           2018-09-04 00:00:00                   1  credit_card   

   payment_installments  payment_value  
0                     1          18.12  
1                     1           2.00  
2                     1          18.59  
3                     1         141.46  
4                     3         179.12

In [ ]:
#Quitamos aquellas columnas que no usaremos
df_merge_orders_payments.drop(['customer_id','order_approved_at','order_delivered_carrier_date','order_delivered_customer_date','order_estimated_delivery_date'], axis=1, inplace=True) #Indicamos el nombre de las columnas y que vamos a mantener los cambios
df_merge_orders_payments.head() #Vemos el resultado

order_id order_status order_purchase_timestamp  \
0  e481f51cbdc54678b7cc49136f2d6af7    delivered      2017-10-02 10:56:33   
1  e481f51cbdc54678b7cc49136f2d6af7    delivered      2017-10-02 10:56:33   
2  e481f51cbdc54678b7cc49136f2d6af7    delivered      2017-10-02 10:56:33   
3  53cdb2fc8bc7dce0b6741e2150273451    delivered      2018-07-24 20:41:37   
4  47770eb9100c2d0c44946d9cf07ec65d    delivered      2018-08-08 08:38:49   

   payment_sequential payment_type  payment_installments  payment_value  
0                   1  credit_card                     1          18.12  
1                   3      voucher                     1           2.00  
2                   2      voucher                     1          18.59  
3                   1       boleto                     1         141.46  
4                   1  credit_card                     3         179.12

In [ ]:
#Exportamos como csv nuestro dataframe
df_merge_orders_payments.to_csv("dataset_generados/orders_payments.csv")

#Order Items - Products

In [ ]:
#Realizamos un merge
df_merge_order_items_products = pd.merge(df_order_items, df_products, how = "inner", on= "product_id")
df_merge_order_items_products.head()


order_id order_item_id  \
0  00010242fe8c5a6d1ba2dd792cb16214             1   
1  00018f77f2f0320c557190d7a144bdd3             1   
2  000229ec398224ef6ca0657da4fc703e             1   
3  00024acbcdf0a6daa1e931b038114c75             1   
4  00042b26cf59d7ce69dfabb4e55b4fd9             1   

                         product_id                         seller_id  \
0  4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   
1  e5f2d52b802189ee658865ca93d83a8f  dd7ddc04e1b6c2c614352b383efe2d36   
2  c777355d18b72b67abbeef9df44fd0fd  5b51032eddd242adc84c38acab88f23d   
3  7634da152a4610f1595efa32f14722fc  9d7a1d34a5052409006425275ba1c2b4   
4  ac6c3623068f30de03045865e4e10089  df560393f3a51e74553ab94004ba5c87   

   shipping_limit_date   price  freight_value product_category_name  \
0  2017-09-19 09:45:35   58.90          13.29            cool_stuff   
1  2017-05-03 11:05:13  239.90          19.93              pet_shop   
2  2018-01-18 14:48:30  199.00          17.87      moveis_decoracao   
3  2018-08-15 10:10:18   12.99          12.79            perfumaria   
4  2017-02-13 13:57:51  199.90          18.14    ferramentas_jardim   

   product_name_lenght  product_description_lenght  product_photos_qty  \
0                 58.0                       598.0                 4.0   
1                 56.0                       239.0                 2.0   
2                 59.0                       695.0                 2.0   
3                 42.0                       480.0                 1.0   
4                 59.0                       409.0                 1.0   

   product_weight_g  product_length_cm  product_height_cm  product_width_cm  
0             650.0               28.0                9.0              14.0  
1           30000.0               50.0               30.0              40.0  
2            3050.0               33.0               13.0              33.0  
3             200.0               16.0               10.0              15.0  
4            3750.0               35.0               40.0              30.0

In [ ]:
#Quitamos aquellas columnas que no usaremos
df_merge_order_items_products.drop(['product_id','seller_id','shipping_limit_date'], axis=1, inplace=True) #Indicamos el nombre de las columnas y que vamos a mantener los cambios
df_merge_order_items_products.head() #Vemos el resultado


order_id order_item_id   price  freight_value  \
0  00010242fe8c5a6d1ba2dd792cb16214             1   58.90          13.29   
1  00018f77f2f0320c557190d7a144bdd3             1  239.90          19.93   
2  000229ec398224ef6ca0657da4fc703e             1  199.00          17.87   
3  00024acbcdf0a6daa1e931b038114c75             1   12.99          12.79   
4  00042b26cf59d7ce69dfabb4e55b4fd9             1  199.90          18.14   

  product_category_name  product_name_lenght  product_description_lenght  \
0            cool_stuff                 58.0                       598.0   
1              pet_shop                 56.0                       239.0   
2      moveis_decoracao                 59.0                       695.0   
3            perfumaria                 42.0                       480.0   
4    ferramentas_jardim                 59.0                       409.0   

   product_photos_qty  product_weight_g  product_length_cm  product_height_cm  \
0                 4.0             650.0               28.0                9.0   
1                 2.0           30000.0               50.0               30.0   
2                 2.0            3050.0               33.0               13.0   
3                 1.0             200.0               16.0               10.0   
4                 1.0            3750.0               35.0               40.0   

   product_width_cm  
0              14.0  
1              40.0  
2              33.0  
3              15.0  
4              30.0

In [ ]:
#Exportamos como csv nuestro dataframe
df_merge_order_items_products.to_csv("dataset_generados/order_items_products.csv")
